src/esp/gfx/WindowlessContext.cpp

```diff
@@ -213,8 +117,8 @@ struct ESPGLXContext : ESPContext {
   int gpuDevice() const { return 0; }
 
  private:
-  Mn::Platform::WindowlessGlxContext glxCtx_;
   Mn::Platform::GLContext magnumGlContext_;
+  Mn::Platform::WindowlessGlxContext glxCtx_;
```

Comment:
And I think this could be collapsed together with the CGL variant for Apple (use `WindowlessGLContext` for both) ... actually maybe even together with EGL, with the same code for all platforms, only have the 

```cpp
    CHECK_EQ(device, 0)
        << "glX context does not support multiple GPUs. Please compile with "
           "BUILD_GUI_VIEWERS=0 for multi-gpu support via EGL";
    CHECK(std::getenv("DISPLAY") != nullptr)
        << "DISPLAY not detected. For headless systems, compile with "
           "--headless for EGL support";
```

checks wrapped in `#ifndef ESP_BUILD_EGL_SUPPORT`? Ultimately, the `virtual`s shouldn't be needed either, I think those would only make sense if there was runtime switching between GLX and EGL -- which there isn't. 

(I'm pretty certain magnum would fail to link if you would attempt to link to both GLX and EGL, although runtime switching between GLX and EGL is still my roadmap. I secretly hope that GLX becomes obsolete before I get to implement that, haha.)